python/src/multigroup/EnergyGroupStructure.python.cpp

```diff
@@ -56,7 +59,7 @@ void wrapEnergyGroupStructure( python::module& module, python::module& ) {
 
     "from_string",
     [] ( const std::string& string, std::size_t number ) -> Record
-       { return read< Record >( string, number ); },
+       { return readWithSubtype< Record >( string, number ); },
```

Comment:
Because the structure has the particle type in the keyword. This particular function reads the particle number and then creates the proper record, which then reads the structure that follows.